### PR TITLE
Addlistener after connect

### PR DESCRIPTION
**Pull request by Aschen**
## What does this PR do?

When someone instantiate Kuzzle, we start to add listener on events.  
With SocketIO protocol, these events must be added also on the `socket` but the `socket` itself is instantiated only after a call to `connect`

### How should this be manually tested?

  - Step 1 : Instantiate Kuzzle with SocketIO protocol

### CODE REVIEW

File: test/kuzzle/constructor.test.js [hunk 1]
```diff
@@ -134,4 +139,22 @@ describe('Kuzzle constructor', () => {
     should(kuzzle.autoQueue).be.true();
     should(kuzzle.autoReplay).be.true();
   });
+
+  it('should initialize kuzzle with SocketIO', () => {
+    new Kuzzle(
+      new SocketIO('localhost')
```
Comment (scottinet) [inline test/kuzzle/constructor.test.js]: what does this test test?

Comment (Aschen) [reply test/kuzzle/constructor.test.js]: Just test that we can instantiate Kuzzle with SocketIO.  
This PR fix an error on Kuzzle instantiation with SocketIO

Comment (scottinet) [reply test/kuzzle/constructor.test.js]: I'm reluctant validating a test that does not include `should` checks. Neither the test title nor its content make it clear what is tested. Especially, I see no correlation between the events being registered at a later time, and the instantiation of a protocol subclass.

Comment (Aschen) [reply test/kuzzle/constructor.test.js]: I just added these tests because from the beginning the SocketIO protocol has never work because of this bug when instantiating the SDK with this protocol.  
I removed it

Comment (scottinet) [reply test/kuzzle/constructor.test.js]: What I meant is that any maintainer in the future would probably remove these tests out of clues of what these tests test.
Removing those is one way to "fix" the issue, but you could also have left a comment, or used a `should` test such as `should(...).not.throw()` or `should(...).instanceOf(...)`.

File: test/kuzzle/constructor.test.js [hunk 2]
```diff
@@ -139,22 +139,4 @@ describe('Kuzzle constructor', () => {
     should(kuzzle.autoQueue).be.true();
     should(kuzzle.autoReplay).be.true();
   });
-
-  it('should initialize kuzzle with SocketIO', () => {
```
Comment (alexandrebouthinon) [inline test/kuzzle/constructor.test.js]: Why did you remove those tests?

File: test/kuzzle/constructor.test.js (hunk 1 above)
Comment (Aschen) [reply test/kuzzle/constructor.test.js]: Got it :)
